test/kuka_kr6_test.cpp

```diff
@@ -0,0 +1,113 @@
+#include <array>
+#include <gtest/gtest.h>
+#include <random>
+
+#include "opw_kinematics/opw_kinematics.h"
+#include "opw_kinematics/opw_parameters_examples.h" // for makeIrb2400_10<double>()
+#include "opw_kinematics/opw_utilities.h"           // for optional checking
+
+template <typename T>
+using Transform = Eigen::Transform<T, 3, Eigen::Affine>;
+
+template <typename T>
+void comparePoses(const Transform<T> & Ta, const Transform<T> & Tb)
+{
+  using Matrix = Eigen::Matrix<T, 3, 3>;
+  using Vector = Eigen::Matrix<T, 3, 1>;
+
+  Matrix Ra = Ta.rotation(), Rb = Tb.rotation();
+  for (int i = 0; i < Ra.rows(); ++i)
+  {
+    for (int j = 0; j < Ra.cols(); ++j)
+    {
+      EXPECT_NEAR(Ra(i, j), Rb(i, j), 1e-6);
+    }
+  }
+
+  Vector pa = Ta.translation(), pb = Tb.translation();
+  EXPECT_NEAR(pa[0], pb[0], 1e-6);
+  EXPECT_NEAR(pa[1], pb[1], 1e-6);
+  EXPECT_NEAR(pa[2], pb[2], 1e-6);
+}
+
+template <typename T>
+void getRandomJointValues(T * q)
+{
+  static std::default_random_engine en{42}; // random engine
+  static std::uniform_real_distribution<T> rg{-M_PI, M_PI}; // random generator
+
+  q[0] = rg(en);
+  q[1] = rg(en);
+  q[2] = rg(en);
+  q[3] = rg(en);
+  q[4] = rg(en);
+  q[5] = rg(en);
+}
+
+TEST(kuka_kr6, random_reachable_poses_double)
+{
+  const int number_of_tests = 10;
+  const auto abb2400 = opw_kinematics::makeIrb2400_10<double>();
```

Comment:
I'm all down for the test, but why is this a "kuka_kr6" test when we're still using the Irb2400 model?